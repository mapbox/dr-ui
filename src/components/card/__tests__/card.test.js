import renderer from 'react-test-renderer';
import { testCases } from './card-test-cases.js';

describe('card', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.noDescription.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.noDescription;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.noImage.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.noImage;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
