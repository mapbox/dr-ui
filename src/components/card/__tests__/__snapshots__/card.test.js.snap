// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card Card with image renders as expected 1`] = `
<a
  className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
  href="url"
>
  <div
    className="relative h120 mb6"
  >
    <div
      className="relative h120 mb12"
      style={
        Object {
          "backgroundImage": "url('./files/simple-map.png')",
          "backgroundPosition": "center",
          "backgroundSize": "100% auto",
        }
      }
    />
  </div>
  <div>
    <div
      className="mb6"
    >
      <div
        className="inline-block mr18 txt-s txt-bold"
      >
        <div
          className="flex-parent flex-parent--center-cross"
        >
          <div
            className="inline-block w6 bg-orange align-middle mr3"
            style={
              Object {
                "borderRadius": "1px",
                "height": "8px",
              }
            }
          />
          <div
            className="inline-block w6 bg-orange align-middle mr3"
            style={
              Object {
                "borderRadius": "1px",
                "height": "8px",
              }
            }
          />
          <div
            className="inline-block w6 bg-gray-light align-middle mr3"
            style={
              Object {
                "borderRadius": "1px",
                "height": "8px",
              }
            }
          />
          <div
            className="inline-block color-orange ml3"
          >
            Intermediate
          </div>
        </div>
      </div>
      <div
        className="inline-block color-gray txt-s txt-bold"
      >
        <svg
          className="events-none icon inline-block align-t"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": 18,
              "width": 18,
            }
          }
        >
          <use
            xlinkHref="#icon-code"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          />
        </svg>
         
        JavaScript
      </div>
    </div>
    <div
      className="mb6"
    >
      My cool card
    </div>
    <div
      className="color-gray color-gray-on-hover"
    >
      This card is very cool.
    </div>
  </div>
</a>
`;

exports[`card Card with no image renders as expected 1`] = `
<a
  className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
  href="url"
>
  
  <div>
    <div
      className="mb6"
    >
      <div
        className="inline-block mr18 txt-s txt-bold"
      >
        <div
          className="flex-parent flex-parent--center-cross"
        >
          <div
            className="inline-block w6 bg-orange align-middle mr3"
            style={
              Object {
                "borderRadius": "1px",
                "height": "8px",
              }
            }
          />
          <div
            className="inline-block w6 bg-orange align-middle mr3"
            style={
              Object {
                "borderRadius": "1px",
                "height": "8px",
              }
            }
          />
          <div
            className="inline-block w6 bg-gray-light align-middle mr3"
            style={
              Object {
                "borderRadius": "1px",
                "height": "8px",
              }
            }
          />
          <div
            className="inline-block color-orange ml3"
          >
            Intermediate
          </div>
        </div>
      </div>
      <div
        className="inline-block color-gray txt-s txt-bold"
      >
        <svg
          className="events-none icon inline-block align-t"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": 18,
              "width": 18,
            }
          }
        >
          <use
            xlinkHref="#icon-code"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          />
        </svg>
         
        JavaScript
      </div>
    </div>
    <div
      className="mb6"
    >
      My cool card
    </div>
    <div
      className="color-gray color-gray-on-hover"
    >
      This card is still cool despite not having a thumbnail.
    </div>
  </div>
</a>
`;

exports[`card Card without description renders as expected 1`] = `
<a
  className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
  href="url"
>
  <div
    className="relative h120 mb6"
  >
    <div
      className="relative h120 mb12"
      style={
        Object {
          "backgroundImage": "url('./files/simple-map.png')",
          "backgroundPosition": "center",
          "backgroundSize": "100% auto",
        }
      }
    />
  </div>
  <div>
    <div
      className="mb6"
    >
      <div
        className="inline-block mr18 txt-s txt-bold"
      >
        <div
          className="flex-parent flex-parent--center-cross"
        >
          <div
            className="inline-block w6 bg-orange align-middle mr3"
            style={
              Object {
                "borderRadius": "1px",
                "height": "8px",
              }
            }
          />
          <div
            className="inline-block w6 bg-orange align-middle mr3"
            style={
              Object {
                "borderRadius": "1px",
                "height": "8px",
              }
            }
          />
          <div
            className="inline-block w6 bg-gray-light align-middle mr3"
            style={
              Object {
                "borderRadius": "1px",
                "height": "8px",
              }
            }
          />
          <div
            className="inline-block color-orange ml3"
          >
            Intermediate
          </div>
        </div>
      </div>
      <div
        className="inline-block color-gray txt-s txt-bold"
      >
        <svg
          className="events-none icon inline-block align-t"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": 18,
              "width": 18,
            }
          }
        >
          <use
            xlinkHref="#icon-code"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          />
        </svg>
         
        JavaScript
      </div>
    </div>
    <div
      className="mb6"
    >
      My cool card without a description
    </div>
  </div>
</a>
`;
